go-controller/pkg/ovn/gateway.go

```diff
@@ -624,160 +769,259 @@ func (oc *DefaultNetworkController) addExternalSwitch(prefix, interfaceID, nodeN
-	intPriority, _ := strconv.Atoi(priority)
-	p := func(item *nbdb.LogicalRouterPolicy) bool {
-		return item.Priority == intPriority
-	}
-	logicalRouterStaticPolicies, err := libovsdbops.FindLogicalRouterPoliciesWithPredicate(oc.nbClient, p)
-	if err != nil {
-		return nil, fmt.Errorf("unable to find logical router policy: %v", err)
+// remove Logical Router Policy on ovn_cluster_router for a specific node.
+// Specify priorities to only delete specific types
+func (gw *GatewayManager) removeLRPolicies(nodeName string, priorities []string) {
+	if len(priorities) == 0 {
+		priorities = []string{types.NodeSubnetPolicyPriority}
 	}
 
-	return logicalRouterStaticPolicies, nil
-}
-
-func (oc *DefaultNetworkController) createPolicyBasedRoutes(match, priority, nexthops string) error {
-	intPriority, _ := strconv.Atoi(priority)
-	lrp := nbdb.LogicalRouterPolicy{
-		Priority: intPriority,
-		Match:    match,
-		Nexthops: []string{nexthops},
-		Action:   nbdb.LogicalRouterPolicyActionReroute,
+	intPriorities := sets.Set[int]{}
+	for _, priority := range priorities {
+		intPriority, _ := strconv.Atoi(priority)
+		intPriorities.Insert(intPriority)
 	}
 
+	managedNetworkName := gw.netInfo.GetNetworkName()
 	p := func(item *nbdb.LogicalRouterPolicy) bool {
-		return item.Priority == lrp.Priority && item.Match == lrp.Match
+		networkName, isSecondaryNetwork := item.ExternalIDs[types.NetworkExternalID]
+		if !isSecondaryNetwork {
+			networkName = types.DefaultNetworkName
+		}
+		if networkName != managedNetworkName {
+			return false
+		}
+		return strings.Contains(item.Match, fmt.Sprintf("%s ", nodeName)) && intPriorities.Has(item.Priority)
 	}
+	err := libovsdbops.DeleteLogicalRouterPoliciesWithPredicate(gw.nbClient, gw.clusterRouterName, p)
+	if err != nil && !errors.Is(err, libovsdbclient.ErrNotFound) {
+		klog.Errorf("Error deleting policies for network %q with priorities %v associated with the node %s: %v", gw.netInfo.GetNetworkName(), priorities, nodeName, err)
+	}
+}
 
-	err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(oc.nbClient, oc.GetNetworkScopedClusterRouterName(), &lrp, p,
-		&lrp.Nexthops, &lrp.Action)
+func (gw *GatewayManager) syncGatewayLogicalNetwork(
+	node *kapi.Node,
+	l3GatewayConfig *util.L3GatewayConfig,
+	hostSubnets []*net.IPNet,
+	hostAddrs []string,
+	clusterSubnets []*net.IPNet,
+	gwLRPIPs []*net.IPNet,
+	isSCTPSupported bool,
+	ovnClusterLRPToJoinIfAddrs []*net.IPNet,
+	externalIPs []net.IP,
+) error {
+	var err error
+
+	enableGatewayMTU := util.ParseNodeGatewayMTUSupport(node)
+
+	err = gw.GatewayInit(
+		node.Name,
+		clusterSubnets,
+		hostSubnets,
+		l3GatewayConfig,
+		isSCTPSupported,
+		gwLRPIPs,
+		ovnClusterLRPToJoinIfAddrs,
+		externalIPs,
+		enableGatewayMTU,
+	)
 	if err != nil {
-		return fmt.Errorf("error creating policy %+v on router %s: %v", lrp, oc.GetNetworkScopedClusterRouterName(), err)
+		return fmt.Errorf("failed to init shared interface gateway: %v", err)
```

Comment:
let's not bring in "shared" and "local" terms anywhere near ovn package... those words are meant for node side gateway things.. this is pretty much a valid switch for local gateway mode as well.. this tripped me as I was debugging something here..